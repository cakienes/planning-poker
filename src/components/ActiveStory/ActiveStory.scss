.activeStory {
    .pageTitle {
        font-size: 22px;
    }
    .storyPoint {
        border: 1px solid #000;
        padding-right: 30px;
        position: relative;
        margin-top: 20px;
        .storyName {
            position: absolute;
            top: -13px;
            background: #fff;
            left: 15px;
        }
        .storyPointList {
            ul {
                margin: 0;
                list-style: none;
                padding: 0;
                li {
                    &.selected {
                        span {
                            border-color: #28a745;
                        }
                    }
                    span {
                        display: inline-block;
                        padding: 15px;
                        margin: 15px;
                        height: 60px;
                        width: 100%;
                        border: 2px solid #000;
                        text-align: center;
                        cursor: pointer;
                    }
                }
            }
        }

        .voteNumber {
            text-align: center;
            margin: 50px;
            font-weight: bold;
        }
    }
}
