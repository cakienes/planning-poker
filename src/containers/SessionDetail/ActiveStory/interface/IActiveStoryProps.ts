import { RouteComponentProps } from 'react-router';
import ISession from '../../../../interfaces/ISession';
import IUserStory from '../../../../interfaces/IUserStory';

export default interface IActiveStoryProps extends RouteComponentProps<any> {
    activeUserStory?: IUserStory;
    selectedSession?: ISession;
    giveStoryPoint: (voterName: string, storyPoint: number) => void;
}
