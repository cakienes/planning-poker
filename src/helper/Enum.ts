export enum UserStoryStatusEnum {
    VOTED = 'VOTED',
    ACTIVE = 'ACTIVE',
    NOT_VOTED = 'NOT_VOTED',
}
