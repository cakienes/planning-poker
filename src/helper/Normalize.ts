const maxLength: any = (max: number) => (value: string) => (value && value.length > max ? value.slice(0, max) : value);

export { maxLength };
