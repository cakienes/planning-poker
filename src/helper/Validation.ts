const required: any = (value: any) => (value ? undefined : 'Please fill');

const onlyNumber: any = (value: any) => {
    return value && value !== '0' && parseInt(value, 10) ? undefined : 'Must be a number';
};

export { required, onlyNumber };
