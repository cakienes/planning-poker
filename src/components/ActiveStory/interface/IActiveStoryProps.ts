import { UserTypeEnum } from '../../../helper/Enum';
import ISession from '../../../interfaces/ISession';
import IUserStory from '../../../interfaces/IUserStory';

export default interface IActiveStoryProps {
    activeUserStory?: IUserStory;
    selectedSession?: ISession;
    giveStoryPoint: (userType: UserTypeEnum, storyPoint?: number) => void;

    type?: UserTypeEnum;

    developerId?: string;
}
