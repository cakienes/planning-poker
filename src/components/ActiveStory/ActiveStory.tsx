import React from 'react';
import { UserTypeEnum } from '../../helper/Enum';
import IVoter from '../../interfaces/IVoter';
import './ActiveStory.scss';
import IActiveStoryProps from './interface/IActiveStoryProps';

const fibonacciNumbers: number[] = [1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 134];
export class ActiveStory extends React.Component<IActiveStoryProps, any> {
    render() {
        const { activeUserStory } = this.props;
        const storyPoint = this.getMyStoryPoint();
        return (
            <div className="activeStory">
                <div className="pageTitle">Active Story</div>
                <div className="storyPoint">
                    <div className="storyName">{activeUserStory && activeUserStory.storyName}</div>
                    <div className="storyPointList">
                        <ul className="row">{this.renderFibonacciNumbers()}</ul>
                    </div>
                    <div className="voteNumber">
                        {storyPoint === -1 ? 'Please Vote!' : (!storyPoint ? '?' : storyPoint) + ' Voted'}
                    </div>
                </div>
            </div>
        );
    }

    renderFibonacciNumbers = (): React.ReactNode => {
        const storyPoint = this.getMyStoryPoint();
        return (
            <ul className="row">
                {fibonacciNumbers.map((x: number) => (
                    <li
                        className={`col-3 ${storyPoint === x ? 'selected' : undefined}`}
                        onClick={() => this.giveStoryPoint(x)}
                        key={x}
                    >
                        <span>{x}</span>
                    </li>
                ))}
                <li className={`col-3 ${!storyPoint ? 'selected' : undefined}`} onClick={() => this.giveStoryPoint()}>
                    <span>?</span>
                </li>
            </ul>
        );
    };

    giveStoryPoint = (value?: number): void => {
        const { type } = this.props;
        if (type) {
            this.props.giveStoryPoint(type, value);
        }
    };

    getMyStoryPoint = (): number => {
        const { activeUserStory, developerId } = this.props;

        if (activeUserStory && activeUserStory.voters) {
            const voter: IVoter | undefined = activeUserStory.voters.find(
                x => x.voterName === (developerId || UserTypeEnum.SCRUM_MASTER),
            );
            if (voter) return voter.storyPoint;
        }
        return -1;
    };
}

export default ActiveStory;
