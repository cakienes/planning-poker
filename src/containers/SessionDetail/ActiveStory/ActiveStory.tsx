import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import IGlobalState from '../../../interfaces/IGlobalState';
import IVoter from '../../../interfaces/IVoter';
import { giveStoryPoint } from '../../../redux/session/session.actions';
import { getActiveUserStory, getSelectedSession } from '../../../redux/session/session.selectors';
import './ActiveStory.scss';
import IActiveStoryProps from './interface/IActiveStoryProps';

const fibonacciNumbers: number[] = [1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 134];
export class ActiveStory extends React.Component<IActiveStoryProps, any> {
    render() {
        const { activeUserStory } = this.props;
        const storyPoint = this.getMyStoryPoint();
        return (
            <div className="activeStory">
                <div className="pageTitle">Active Story</div>
                <div className="storyPoint">
                    <div className="storyName">{activeUserStory && activeUserStory.storyName}</div>
                    <div className="storyPointList">
                        <ul className="row">{this.renderFibonacciNumbers()}</ul>
                    </div>
                    <div className="voteNumber">
                        {!storyPoint ? 'Please Vote!' : (storyPoint === 1000 ? '?' : storyPoint) + ' Voted'}
                    </div>
                </div>
            </div>
        );
    }

    renderFibonacciNumbers = (): React.ReactNode => {
        const storyPoint = this.getMyStoryPoint();
        return (
            <ul className="row">
                {fibonacciNumbers.map((x: number) => (
                    <li
                        className={`col-3 ${storyPoint === x ? 'selected' : undefined}`}
                        onClick={() => this.giveStoryPoint(x)}
                        key={x}
                    >
                        <span>{x}</span>
                    </li>
                ))}
                <li
                    className={`col-3 ${storyPoint === 1000 ? 'selected' : undefined}`}
                    onClick={() => this.giveStoryPoint(1000)}
                >
                    <span>?</span>
                </li>
            </ul>
        );
    };

    giveStoryPoint = (value: number): void => {
        const { match } = this.props;
        if (match && match.params && match.params.voteFor) {
            this.props.giveStoryPoint(match.params.voteFor, value);
        }
    };

    getMyStoryPoint = (): number | string | undefined => {
        const {
            activeUserStory,
            match: {
                params: { voteFor },
            },
        } = this.props;
        if (activeUserStory && activeUserStory.voters && voteFor) {
            const voter: IVoter | undefined = activeUserStory.voters.find(x => x.voterName === voteFor.toUpperCase());
            if (voter) return voter.storyPoint;
            return undefined;
        }
        return -1;
    };
}

const mapStateToProps = (state: IGlobalState) => ({
    activeUserStory: getActiveUserStory(state),
    selectedSession: getSelectedSession(state),
});

export const mapDispatchToProps = (dispatch: Function) => {
    return {
        giveStoryPoint: (voterName: string, storyPoint: number) => {
            dispatch(giveStoryPoint(voterName, storyPoint));
        },
    };
};

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps,
    )(ActiveStory),
);
