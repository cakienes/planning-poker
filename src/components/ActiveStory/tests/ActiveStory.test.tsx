import React from 'react';
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow';
import { UserTypeEnum } from '../../../helper/Enum';
import ActiveStory from '../ActiveStory';
import IActiveStoryProps from '../interface/IActiveStoryProps';

describe('ActiveStory.tsx', () => {
    let renderer: ShallowRenderer;

    const giveStoryPointMock = jest.fn();
    let activeStoryProps: IActiveStoryProps = { giveStoryPoint: giveStoryPointMock };

    beforeEach(() => {
        renderer = createRenderer();

        activeStoryProps = {
            ...activeStoryProps,
            activeUserStory: undefined,
            developerId: undefined,
            selectedSession: undefined,
            type: undefined,
        };
    });

    it('renders correctly ActiveStory', async () => {
        expect(renderer.render(<ActiveStory {...activeStoryProps} />)).toMatchSnapshot();
    });

    it('renders correctly while activeUserStory', async () => {
        activeStoryProps = {
            ...activeStoryProps,
            activeUserStory: {
                storyName: 'storyName',
            },
        };
        expect(renderer.render(<ActiveStory {...activeStoryProps} />)).toMatchSnapshot();
    });

    it('renders correctly while this.getMyStoryPoint return undefined', async () => {
        renderer.render(<ActiveStory {...activeStoryProps} />);
        const instance: any = renderer.getMountedInstance();
        instance.getMyStoryPoint = jest.fn().mockReturnValue(undefined);

        expect(instance.render()).toMatchSnapshot();
    });

    it('renders correctly while this.getMyStoryPoint return 1', async () => {
        renderer.render(<ActiveStory {...activeStoryProps} />);
        const instance: any = renderer.getMountedInstance();
        instance.getMyStoryPoint = jest.fn().mockReturnValue(1);

        expect(instance.render()).toMatchSnapshot();
    });

    it('giveStoryPoint of ActiveStory while tpye is undefined', async () => {
        renderer.render(<ActiveStory {...activeStoryProps} />);
        const instance: any = renderer.getMountedInstance();

        instance.giveStoryPoint();

        expect(giveStoryPointMock).not.toHaveBeenCalled();
    });

    it('giveStoryPoint of ActiveStory while tpye is SCRUM_MASTER', async () => {
        activeStoryProps = {
            ...activeStoryProps,
            type: UserTypeEnum.SCRUM_MASTER,
        };
        renderer.render(<ActiveStory {...activeStoryProps} />);
        const instance: any = renderer.getMountedInstance();

        instance.giveStoryPoint('value');

        expect(giveStoryPointMock).toHaveBeenCalledWith(UserTypeEnum.SCRUM_MASTER, 'value');
    });

    it('getMyStoryPoint of ActiveStory while activeUserStory is undefined', async () => {
        renderer.render(<ActiveStory {...activeStoryProps} />);
        const instance: any = renderer.getMountedInstance();

        expect(instance.getMyStoryPoint()).toEqual(-1);
    });

    it('getMyStoryPoint of ActiveStory while activeUserStory is not undefined & activeUserStory.voters is undefined', async () => {
        activeStoryProps = {
            ...activeStoryProps,
            activeUserStory: {
                storyName: 'storyName',
            },
        };
        renderer.render(<ActiveStory {...activeStoryProps} />);
        const instance: any = renderer.getMountedInstance();

        expect(instance.getMyStoryPoint()).toEqual(-1);
    });

    it('getMyStoryPoint of ActiveStory while voters empty', async () => {
        activeStoryProps = {
            ...activeStoryProps,
            activeUserStory: {
                storyName: 'storyName',
                voters: [
                    {
                        voterName: 'developerId',
                        storyPoint: 1,
                    },
                ],
            },
            developerId: 'developerId',
        };
        renderer.render(<ActiveStory {...activeStoryProps} />);
        const instance: any = renderer.getMountedInstance();

        expect(instance.getMyStoryPoint()).toEqual(1);
    });

    it('getMyStoryPoint of ActiveStory while voters empty', async () => {
        activeStoryProps = {
            ...activeStoryProps,
            activeUserStory: {
                storyName: 'storyName',
                voters: [
                    {
                        voterName: UserTypeEnum.SCRUM_MASTER,
                        storyPoint: 2,
                    },
                ],
            },
        };
        renderer.render(<ActiveStory {...activeStoryProps} />);
        const instance: any = renderer.getMountedInstance();

        expect(instance.getMyStoryPoint()).toEqual(2);
    });
});
