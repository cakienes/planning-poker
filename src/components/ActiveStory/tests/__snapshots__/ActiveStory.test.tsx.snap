// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveStory.tsx renders correctly ActiveStory 1`] = `
<div
  className="activeStory"
>
  <div
    className="pageTitle"
  >
    Active Story
  </div>
  <div
    className="storyPoint"
  >
    <div
      className="storyName"
    />
    <div
      className="storyPointList"
    >
      <ul
        className="row"
      >
        <ul
          className="row"
        >
          <FibonacciBox
            fibonacciNumber={1}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={2}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={3}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={5}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={8}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={13}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={21}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={34}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={55}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={89}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={134}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={0}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
        </ul>
      </ul>
    </div>
    <div
      className="voteNumber"
    >
      Please Vote!
    </div>
  </div>
</div>
`;

exports[`ActiveStory.tsx renders correctly while activeUserStory 1`] = `
<div
  className="activeStory"
>
  <div
    className="pageTitle"
  >
    Active Story
  </div>
  <div
    className="storyPoint"
  >
    <div
      className="storyName"
    >
      storyName
    </div>
    <div
      className="storyPointList"
    >
      <ul
        className="row"
      >
        <ul
          className="row"
        >
          <FibonacciBox
            fibonacciNumber={1}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={2}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={3}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={5}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={8}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={13}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={21}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={34}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={55}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={89}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={134}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
          <FibonacciBox
            fibonacciNumber={0}
            giveStoryPoint={[Function]}
            storyPoint={-1}
          />
        </ul>
      </ul>
    </div>
    <div
      className="voteNumber"
    >
      Please Vote!
    </div>
  </div>
</div>
`;

exports[`ActiveStory.tsx renders correctly while this.getMyStoryPoint return 1 1`] = `
<div
  className="activeStory"
>
  <div
    className="pageTitle"
  >
    Active Story
  </div>
  <div
    className="storyPoint"
  >
    <div
      className="storyName"
    />
    <div
      className="storyPointList"
    >
      <ul
        className="row"
      >
        <ul
          className="row"
        >
          <FibonacciBox
            fibonacciNumber={1}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
          <FibonacciBox
            fibonacciNumber={2}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
          <FibonacciBox
            fibonacciNumber={3}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
          <FibonacciBox
            fibonacciNumber={5}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
          <FibonacciBox
            fibonacciNumber={8}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
          <FibonacciBox
            fibonacciNumber={13}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
          <FibonacciBox
            fibonacciNumber={21}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
          <FibonacciBox
            fibonacciNumber={34}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
          <FibonacciBox
            fibonacciNumber={55}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
          <FibonacciBox
            fibonacciNumber={89}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
          <FibonacciBox
            fibonacciNumber={134}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
          <FibonacciBox
            fibonacciNumber={0}
            giveStoryPoint={[Function]}
            storyPoint={1}
          />
        </ul>
      </ul>
    </div>
    <div
      className="voteNumber"
    >
      1 Voted
    </div>
  </div>
</div>
`;

exports[`ActiveStory.tsx renders correctly while this.getMyStoryPoint return undefined 1`] = `
<div
  className="activeStory"
>
  <div
    className="pageTitle"
  >
    Active Story
  </div>
  <div
    className="storyPoint"
  >
    <div
      className="storyName"
    />
    <div
      className="storyPointList"
    >
      <ul
        className="row"
      >
        <ul
          className="row"
        >
          <FibonacciBox
            fibonacciNumber={1}
            giveStoryPoint={[Function]}
          />
          <FibonacciBox
            fibonacciNumber={2}
            giveStoryPoint={[Function]}
          />
          <FibonacciBox
            fibonacciNumber={3}
            giveStoryPoint={[Function]}
          />
          <FibonacciBox
            fibonacciNumber={5}
            giveStoryPoint={[Function]}
          />
          <FibonacciBox
            fibonacciNumber={8}
            giveStoryPoint={[Function]}
          />
          <FibonacciBox
            fibonacciNumber={13}
            giveStoryPoint={[Function]}
          />
          <FibonacciBox
            fibonacciNumber={21}
            giveStoryPoint={[Function]}
          />
          <FibonacciBox
            fibonacciNumber={34}
            giveStoryPoint={[Function]}
          />
          <FibonacciBox
            fibonacciNumber={55}
            giveStoryPoint={[Function]}
          />
          <FibonacciBox
            fibonacciNumber={89}
            giveStoryPoint={[Function]}
          />
          <FibonacciBox
            fibonacciNumber={134}
            giveStoryPoint={[Function]}
          />
          <FibonacciBox
            fibonacciNumber={0}
            giveStoryPoint={[Function]}
          />
        </ul>
      </ul>
    </div>
    <div
      className="voteNumber"
    >
      ? Voted
    </div>
  </div>
</div>
`;
